### PR TITLE
feat!: bump version

### PR DESCRIPTION
**Pull request by shkm**
Does a bunch of things, see commit messages :)

### CODE REVIEW

**Review by coddeys — COMMENTED** [review]
LGTM. What about adding an `hc-` prefix to all "new" gems

**Comment by shkm** [comment]
> LGTM. What about adding an `hc-` prefix to all "new" gems

Perhaps internal gems, but this is staying public because it's much easier -- allows us to just push it to rubygems :)

**Review by coddeys — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## HEAD (unreleased)
 
+## 4.0 (28th Feb 2023)
+Authors: Jamie Schembri
+
+* feature: rename gem from `wisper` to `wisper-compat`.
```
Comment (jmgarnier) [inline CHANGELOG.md:6]: Rename does say much about the plan. README is very vague to me 

(Decouple core business logic from external concerns in Hexagonal style architectures, Use as an alternative to ActiveRecord callbacks and Observers in Rails apps, Pub / Sub ...) 

Can you elaborate a bit on your plans?

File: wisper-compat.gemspec [hunk 2]
```diff
@@ -4,29 +4,26 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'wisper/version'
 
 Gem::Specification.new do |gem|
-  gem.name          = "wisper"
+  gem.name          = "wisper-compat"
   gem.version       = Wisper::VERSION
-  gem.authors       = ["Kris Leech"]
-  gem.email         = ["kris.leech@gmail.com"]
+  gem.authors       = ["Kris Leech", "Dima Sukhikh", "Jamie Schembri"]
+  gem.email         = ["kris.leech@gmail.com", "dima.sukhikh@nedap.com", "jamie.schembri@nedap.com"]
```
[the remark this answers is not in the document]
Comment (shkm) [reply wisper-compat.gemspec:10]: It's public

File: wisper-compat.gemspec [hunk 3]
```diff
@@ -4,29 +4,26 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
   gem.description   = <<-DESC
     A micro library providing objects with Publish-Subscribe capabilities.
     Both synchronous (in-process) and asynchronous (out-of-process) subscriptions are supported.
     Check out the Wiki for articles, guides and examples: https://github.com/krisleech/wisper/wiki
```
[the remark this answers is not in the document]
Comment (shkm) [reply wisper-compat.gemspec:14]: Yeah, but forks don't seem to copy wiki :/.

File: CHANGELOG.md (hunk 1 above)
Comment (shkm) [reply CHANGELOG.md:6]: Remain compatible with Ruby :)

File: wisper-compat.gemspec (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmgarnier) [reply wisper-compat.gemspec:10]: woopsy!

**Pull request merged**